src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -257,39 +279,66 @@ void damped_harmonic_impl(
     get<0>(d4_RW_L2) += get(weight) * d0_roll_on;
   }
 
+  time_deriv_of_lapse<DataVector, SpatialDim, Frame>(
+      make_not_null(&dt_lapse), lapse, shift, spacetime_unit_normal, phi, pi);
```

Comment:
Here you can use `lapse * (lapse * half_pi_two_normals - shift^i half_phi_two_normals_i)` to eliminate all the normal contractions work that otherwise is done